x-pack/plugins/cloud_security_posture/server/create_indices/create_indices.ts

```diff
@@ -140,6 +151,64 @@ const createLatestFindingsIndex = async (esClient: ElasticsearchClient, logger:
   }
 };
 
+const createLatestVulnerabilitiesIndex = async (esClient: ElasticsearchClient, logger: Logger) => {
+  try {
+    // Deletes old assets from previous versions as part of upgrade process
+    const INDEX_TEMPLATE_V830 = 'cloud_security_posture.vulnerabilities_latest';
+    await deleteIndexTemplateSafe(esClient, logger, INDEX_TEMPLATE_V830);
+
+    // We want that our latest findings index template would be identical to the findings index template
+    const vulnerabilitiesIndexTemplateResponse = await esClient.indices.getIndexTemplate({
+      name: VULNERABILITIES_INDEX_NAME,
+    });
+
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { template, composed_of, _meta } =
+      vulnerabilitiesIndexTemplateResponse.index_templates[0].index_template;
+
+    // We always want to keep the index template updated
+    await esClient.indices.putIndexTemplate({
+      name: LATEST_VULNERABILITIES_INDEX_TEMPLATE_NAME,
+      index_patterns: LATEST_VULNERABILITIES_INDEX_PATTERN,
+      priority: 500,
+      template: {
+        mappings: template?.mappings,
+        settings: {
+          ...template?.settings,
+          default_pipeline: latestVulnerabilitiesPipelineIngestConfig.id,
+          lifecycle: {
+            name: '',
+          },
+        },
+        aliases: template?.aliases,
+      },
+      _meta,
+      composed_of,
+    });
+
+    const result = await createIndexSafe(esClient, logger, LATEST_VULNERABILITIES_INDEX_DEFAULT_NS);
+
+    if (result === 'already-exists') {
+      // Make sure mappings are up-to-date
+      const simulateResponse = await esClient.indices.simulateTemplate({
+        name: LATEST_VULNERABILITIES_INDEX_TEMPLATE_NAME,
+      });
+
+      await updateIndexSafe(
+        esClient,
+        logger,
+        LATEST_VULNERABILITIES_INDEX_DEFAULT_NS,
+        simulateResponse.template.mappings
+      );
+    }
+  } catch (e) {
+    logger.error(
+      `Failed to upsert index template [Template: ${LATEST_VULNERABILITIES_INDEX_TEMPLATE_NAME}]`
+    );
+    logger.error(e);
+  }
+};
+
```

Comment:
Today, we have two methods for creating indices, one for the score index and the other for the latest findings index, due to their different logics. 
Kibana maintains the index template for the score index, the latest findings index fetches the template from the findings index (which is maintained in the integration) and then creates the latest findings index. 

Creating the vulnerability index follows the same logic as creating the latest findings index, right? 
In this case, why not have one generic function that works with parameters and can serve both cases?

Reply:
That's a good catch. I'll create one generic function.